46.md

```diff
@@ -1,98 +1,174 @@
+   - The remote pubkey is the pubkey that the user is trying to sign in / sign as. This can be fetched with an NIP-05 or entered directly by the user.
+2. Clients use the local keypair for encrypting request events that they send to remote signers.
+
+### Remote signer-side
+
+1. Remote signers manage the private key of the user that will be used for signing. There are many different remote signers out there (e.g. Nsecbunker, Amber, etc.) that operate in different ways and have different features.
+2. Remote signers wait for request events and then act.
+
+### Example flow for signing an event
+
+Coming soon...
+
+### Example Oauth-like flow to create a new user account with Nsecbunker
+
+Coming soon...
 
-## Event payloads
+## Request Events `Multiple kinds`
 
-Event payloads are [NIP-04](04.md)-encrypted JSON blobs that look like JSONRPC messages (their format is specified inside the `.content` of the event formats below).
+Request events take the following form. In order simplify requests and payloads as much as possible, there are different event kinds for each command which help remote signers to understand the request and process the content properly.
 
-Events sent by the client to the remote signer have the following format:
+Clients can subscribe to events of `kind: 24136`, `kind: 24137`, and `kind: 24138` that have an "e" tag containing the ID of their request event to listen for the response, auth_challenge, or error to their command.
 
-```js
+```json
 {
-  "pubkey": "<client-key-hex>"
-  "kind": 24133,
-  "tags": [
-    ["p", "<signer-key-hex>"]
-  ],
-  "content": "nip04_encrypted_json({id: <random-string>, method: <see-below>, params: [array_of_strings]})",
-  ...
+    "id": <id>,
+    "kind": <kind for command>, // Each command has it's own kind
+    "pubkey": <local pubkey of the user>,
+    "content": <nip44(<request>)>,
+    "tags": [["p", <remote pubkey>]],
+    "created_at": <unix timestamp in seconds>,
 }
 ```
 
-And the events the remote signer sends to the client have the following format:
+| Kind    | Command                  | Content field (always NIP-44 encrypted)                                         | Data Type               |
+| ------- | ------------------------ | ------------------------------------------------------------------------------- | ----------------------- |
+| `24140` | `create_account`         | {username: <username>, domain: <domain>, <any other arbitrary key/value pairs>} | JSON stringified object |
+| `24141` | `connect`                | -                                                                               | null                    |
+| `24142` | `sign_event`             | nostr event to sign                                                             | JSON stringified object |
```

Comment:
The "command" name like `sign_event` isn't actually used by the protocol anymore, right? So in the table you can just write an English statement like "Sign event"

Reply:
That's true. I kept the old command name here to help people make the connection that these new events are the same as the old commands.